### PR TITLE
Some artifacts issues resolved

### LINKED ISSUE #149: Add Minimalist Artifact Listing to Column Options for Monsters
Right now, what I want to know are:

1. How many of my 6* monsters don't have artifacts?
2. How many of them aren't powered up to 15?
3. How many of them are the wrong sort of artifact because I threw something on early on and I need to change them?

There's no easy way to see it (in bulk) either in the game's UI or in Swarfarm's.

It'd be nice to have something like you've got for the runes, where hovering over gives more detail, but you can see it en masse for all your monsters.

FWIW, when I'm looking at this, I'm usually filtering based on 6*, so I'm looking at archetype (because some of the archetypes don't match their skillsets, why com2us why?) leader skill, skills, runes, stats. Having the artifacts next to the runes but before the stats detail would IMHO be ideal, though others may disagree.

### LINKED ISSUE #154: Artifacts: Group by att/hp/def within element/attribute
Most of the time, it seems that people want attack artifacts, just looking at what people have equipped, YouTube, etc. So the current sorting strictly by element/attribute and power up doesn't really make it obvious what's available if you're looking for how many attack fire artifacts you may have.

### LINKED ISSUE #155: Artifacts: Add at-a-glance summary of unassigned artifacts
The way the Com2us UI works, it defaults to hiding the artifacts not relevant to the current monster. This means that it's  easy to accumulate a bunch of, oh, dark support HP artifacts if you farm a lot, and suddenly you can run out of artifact space. Been there.

I'd like to have a small table of unassigned attribute/type artifacts by stat (just the number of them). It probably makes the most sense to put it above the artifact listing (on the artifacts page), but below the filters.

Thoughts?

### LINKED ISSUE #171: Artifacts not adding their main stat to total stats
When looking at a monster's stats, the total stats do NOT reflect the stats added by artifacts. I have two current examples:
WhatsApp Image 2020-11-09 at 01 11 55
image
WhatsApp Image 2020-11-09 at 01 11 56
image

### LINKED ISSUE #172: Server error in Artifact.clean() when adding an artifact
Exception: 
```
TypeError at /profile/...username.../artifacts/add/
'<' not supported between instances of 'NoneType' and 'float'
```

LOC: https://github.com/PeteAndersen/swarfarm/blob/master/bestiary/models/artifacts.py#L384

POST data:
```
slot = '1'
element = 'wind'
archetype = ''
original_quality = '3'
main_stat = '3'
level = '15'
effects_0 = '1'
effects_1 = '20'
effects_2 = '19'
effects_3 = '13'
effects_value_0 = '15'
effects_value_1 = '4'
effects_value_2 = ''
effects_value_3 = '3'
assigned_to = ...removed...
```

Error caused by effects_value_2 being skipped over in the form. This should raise a ValidationError instead so the error propagates to the form.

### PR DESCRIPTION
**Pull request by QuatZo**
* Fixed Artifact group counter showing "0 runes" instead of "<num> artifacts"
* Fixed Artifact page title (Runes -> Artifacts)
* Added Artifacts to Monster List
* Added Artifacts to Monster Popover
* Fixed Artifact Popover not having element/archetype logo
* Added Unknown image (couldn't find in static files)
* Added nested groups (by mainstat) in Artifact slot groups

resolves #149 
resolves #154 
resolves #155 
resolves #171 
resolves #172

### CODE REVIEW

File: herders/templates/herders/profile/artifacts/inventory.html [hunk 1]
```diff
@@ -48,19 +48,49 @@
...
                                     <strong>{{ group.name }}</strong>
                                 </a>
-                                <span class="pull-right">{% with num_runes=group.runes|length %}{{ num_runes }} rune{{ num_runes|pluralize }}{% endwith %}</span>
+                                <span class="pull-right">{% with num_runes=group.artifacts|length %}{{ num_runes }} artifact{{ num_runes|pluralize }}{% endwith %}</span>
                             </div>
                             <div id="runegroup{{ forloop.counter }}" class="collapse in">
                                 <div class="panel-body">
+                                    <!-- If groupped by slot, then group them internally by main stat -->
+                                    {% if box_grouping == 'slot' %}
+                                    <div class="panel-body">
```
Comment (PeteAndersen) [inline herders/templates/herders/profile/artifacts/inventory.html]: This `panel-body` is nested in the `panel-body` div from L54, causing extra padding:

image

**Review by PeteAndersen — CHANGES_REQUESTED** [review]

**Comment by PeteAndersen** [comment]
Looks overall great, but will wait for @deirdresm to have a chance to look at it too before merging

**Comment by QuatZo** [comment]
Deleted additional padding
image

**Comment by QuatZo** [comment]
I've checked it as a draft since there are some other issues with artifacts I would like to resolve before merging it to master:

- [x] https://swarfarm.com/feedback/143/
- [x] https://swarfarm.com/feedback/144/
- [x] Issue #171

**Comment by PeteAndersen** [comment]
Needs a migration file committed for the change from IntegerField to FloatField. Will do a full review in a few hours.

**Comment by QuatZo** [comment]
Commit https://github.com/PeteAndersen/swarfarm/pull/165/commits/ce6f529135147be64d94f2074f8057f91eeb7380 resolves #171 
Commit https://github.com/PeteAndersen/swarfarm/pull/165/commits/dba30335ef348b7795d4fafc92bf192f7f5f9103 resolves #172

**Comment by PeteAndersen** [comment]
All good except one minor issue: Monster stats table doesn't update when adding/removing an artifact on the monster view page. This is handled in `herders/static/herders/js/monster_view.js`, needs a call to `UpdateStats()` added to the `AssignArtifactChoice()` and `UnassignArtifact()` methods.

**Comment by QuatZo** [comment]
> All good except one minor issue: Monster stats table doesn't update when adding/removing an artifact on the monster view page. This is handled in `herders/static/herders/js/monster_view.js`, needs a call to `UpdateStats()` added to the `AssignArtifactChoice()` and `UnassignArtifact()` methods.

Fixed, works like a charm

**Review by PeteAndersen — APPROVED** [review]

**Pull request merged**